core/arch/arm/kernel/boot.c

```diff
@@ -1152,6 +1152,7 @@ static void init_primary(unsigned long pageable_part, unsigned long nsec_entry)
 	 */
 	thread_get_core_local()->curr_thread = 0;
 	init_runtime(pageable_part);
+	thread_get_core_local()->curr_thread = -1;
```

Comment:
This is only needed with `CFG_VIRTUALIZATION` and `thread_init_boot_thread()` immediately undoes this.
Perhaps we could do it like:
``` C
#ifdef CFG_VIRTUALIZATION
	/* A nice comment explaining why this is needed */
	thread_get_core_local()->curr_thread = -1;
#else
	thread_init_boot_thread();
#endif

Reply:
`if IS_ENABLED(CFG_VIRTUALIZATION)` would work in this case